.github/workflows/nm-build-test.yml

```diff
@@ -123,37 +166,236 @@ jobs:
     #         python: ${{ inputs.python }}
     #         whl: ${{ needs.BUILD.outputs.whl }}
     #         test_skip_env_vars: ${{ inputs.test_skip_env_vars }}
+    #         test_directory: distributed
     #     secrets: inherit
+    
+    ENGINE:
+        needs: [BUILD]
+        if: success()
+        uses: ./.github/workflows/nm-test.yml
+        with:
+            test_label: ${{ inputs.test_label_solo }}
+            timeout: ${{ inputs.test_timeout }}
+            gitref: ${{ github.ref }}
+            python: ${{ inputs.python }}
+            whl: ${{ needs.BUILD.outputs.whl }}
+            test_skip_env_vars: ${{ inputs.test_skip_env_vars }}
+            test_directory: engine
+        secrets: inherit
 
-    BENCHMARK:
+    ENTRYPOINTS:
+        needs: [BUILD]
+        if: success()
+        uses: ./.github/workflows/nm-test.yml
+        with:
+            test_label: ${{ inputs.test_label_solo }}
+            timeout: ${{ inputs.test_timeout }}
+            gitref: ${{ github.ref }}
+            python: ${{ inputs.python }}
+            whl: ${{ needs.BUILD.outputs.whl }}
+            test_skip_env_vars: ${{ inputs.test_skip_env_vars }}
+            test_directory: entrypoints
+        secrets: inherit
+  
+    KERNELS:
```

Comment:
we should make the "fan out" dynamic so we can drive it via input parameter.

Reply:
sg - ill let you finish off the PR